### PR TITLE
Add iteration support for touchpads

### LINKED ISSUE #103: Capacitive touch iterable attribute
This code was designed in response to [this issue](https://github.com/adafruit/Adafruit_CircuitPython_CircuitPlayground/issues/12).

It is a start for another library that could be for the Circuit Playground Express, or usable across multiple boards. I'm posting the code here so anyone who wants to pick this up has a place to start. The idea is to allow for iteration across all capacitive touch inputs (in the current case, the touch pads on the CPX), and return strings, ints or pin instances for use. It could possibly be written into a class that uses touch state. Or it could be expandable to include all inputs that provide boolean values, such as buttons as well.

As it is, it initialises all of the touch pads as touch pads, which on the CPX means that none of those pads can be used anything else, such as digital outputs, at the same time that this function is called. This may not be an issue in the eventual library, but it's something to be aware of.

The functions are:
```
def touched_padname(self):
    pin = (board.A1, board.A2, board.A3, board.A4, board.A5, board.A6, board.A7)
    touch = tuple(touchio.TouchIn(p) for p in pin)
    return ['A{}'.format(i+1) for (i, v) in enumerate(touch) if v.value]

def touched(self):
    pin = (board.A1, board.A2, board.A3, board.A4, board.A5, board.A6, board.A7)
    touch = tuple(touchio.TouchIn(p) for p in pin)
    return [i for (i, v) in enumerate(touch) if v.value]
```
This allows for use like the following examples. It is by no means limited to the simplicity of these examples. 
```
for pad in cpx.touched:
    print("pad {0} touched!".format(pad))
```
```
if 1 in cpx.touched:
    print("pad 1 touched!")
```
```
if 'A1' in cpx.touched:
    print("pad A1 touched!")
```
```
COMBO = ('A1', 'A2', 'A5')
if all(pad in cpx.touched for pad in COMBO):
    print("UNLOCKED!")
```
Opened here from original issue [here](https://github.com/adafruit/circuitpython/issues/367).

### PR DESCRIPTION
**Pull request by tekktrik**
Resolves #103 as follows:

1. Adds `touched` property for showing list of of pin names currently registering a touch
2. Adds `touchpads` property got showing list of pin names that are set up as touchpad inputs
3. Continues to allow inputs to only be initialized as `touchio.TouchIn` as needed
4. Now allows for touchpads to be initialized and deinitialized as touchpad inputs, which allows those pins to be used for other things if wanted!
5. Existing methods for checking touchpads still works!
6. Modifies/adds examples to demonstrate new functionality

This is all powered by the new `IterableInput` class, which should remain hidden from the user but provides the above functionalities.

### CODE REVIEW

**Review by kattni — CHANGES_REQUESTED** [review]
My concern here is increasing the size of the library. We need to make sure it still fits frozen in CircuitPython for the CPX before considering merging this.

**Comment by tekktrik** [comment]
Sounds good, I can look at removing anything not immediately needed as well.  If any of the above functionalities aren't needed, let me know an I can try and factor them out.  Also, if there's a good way to test this library as frozen.

**Comment by kattni** [comment]
@tekktrik There absolutely is a way to test it! That's the next step before trying to slim down the code. Do you know how to build CircuitPython? Testing a frozen library is pretty simple once you have a CircuitPython build environment set up.

**Comment by tekktrik** [comment]
No, I don't actually, I think I've seen guides though, is there a Learn guide for it?

**Comment by kattni** [comment]
Yep! It's [here](https://learn.adafruit.com/building-circuitpython/build-circuitpython). Please message me on Discord if you need assistance. Once you have the build environment working, I'll show you how to test the frozen library.

**Comment by tekktrik** [comment]
So it builds, and it currently doesn't look like I have too much in the way of "unused" code.  So I think it really depends on what functionalities are worth introducing and just cutting the rest:

1. `touched` property for showing pin names (`str`) currently **registering touch**
2. `touchpads` shows the pins names (`str`) that are currently **setup as touch inputs**
3. It only initializes touch inputs as needed, similar to how it's been done before this
4. You can init and deinit touchpads.  This seemed like a big want, but I don't know the balance of freezing this into the firmware vs trying to recover the pins as touchpads or saving RAM
5. You can access pins by both pin names (`str`) and pin numbers (`int`)

Let me know what to cutout!

**Comment by tekktrik** [comment]
Also, it may make sense to hide this class in the documentation, let me know if I should exclude it in `api.rst`

File: adafruit_circuitplayground/circuit_playground_base.py [hunk 1]
```diff
@@ -50,6 +50,95 @@ def light(self):
...
+        return input_pad
+
+    @staticmethod
+    def _use_str_name(value):
+        """Converts an index into the pin name if needed"""
+
+        if value in (7, "A7", "TX"):
+            return "TX"
+        if isinstance(value, int):
+            if value not in range(1, 7):
```
[the remark this answers is not in the document]
Comment (tekktrik) [reply adafruit_circuitplayground/circuit_playground_base.py:97]: Oh neat, I didn't think about that, thanks!

**Comment by tekktrik** [comment]
Thanks for reviewing!  Let me take a stab at this, but I may end up just creating a new branch for this if it makes more sense to start from scratch :)

**Comment by kattni** [comment]
@tekktrik Starting a new PR is perfectly fine. Simply make sure you reference this PR in the new PR notes so there's some continuity there. You can leave this open until you create the new one, if you like.

**Pull request closed without merging**